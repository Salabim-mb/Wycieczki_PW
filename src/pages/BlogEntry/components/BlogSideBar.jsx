import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import { Breadcrumbs, Chip, Grid, Paper, Typography } from '@material-ui/core';
import NavigateNextIcon from '@material-ui/icons/NavigateNext';
import PropTypes from 'prop-types';

const useStyles = makeStyles((theme) => ({
	wrapper: {
		maxWidth: '100%',
		padding: theme.spacing(2, 0)
	},
	sidebarBreadcrumbs: {
		padding: theme.spacing(2),
		backgroundColor: theme.palette.grey[200],
		margin: theme.spacing(4, 0, 3, 0)
	},
	sidebarSection: {
		marginTop: theme.spacing(3),
	},
}));

const BlogSideBar = ({pathParams, featuredPosts, setRedirect}) => {
	const classes = useStyles();

	return (
		<Grid item xs={12} md={4} className={classes.wrapper}>
			<Paper elevation={0} className={classes.sidebarBreadcrumbs}>
				<Typography variant="h6" component="h6">
					Nawigacja
				</Typography>
				<Breadcrumbs separator={<NavigateNextIcon fontSize="small" />} aria-label="breadcrumb">
					{pathParams.map((item) => (
						<Chip
							key={item.path}
							label={item.name}
							variant="outlined"
							onClick={() => setRedirect({action: true, destination: item.path})}
						/>
					))}
				</Breadcrumbs>
			</Paper>
			<Typography variant="h5" component="h6">
				Proponowane wpisy
			</Typography>
			Tu będą ze 3 tytuły proponowanych / losowych wpisów
			{featuredPosts}
		</Grid>
	);
};

BlogSideBar.propTypes = {
	pathParams: PropTypes.arrayOf(
		PropTypes.objectOf({
			name: PropTypes.string,
			path: PropTypes.oneOfType([PropTypes.string, PropTypes.func])
		})
	).isRequired,
	featuredPosts: PropTypes.arrayOf(
		null // instancja posta
	).isRequired,
	setRedirect: PropTypes.func.isRequired
}

export default BlogSideBar;