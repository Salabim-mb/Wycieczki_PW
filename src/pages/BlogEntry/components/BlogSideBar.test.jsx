import React from 'react';
import {render, fireEvent} from '@testing-library/react';
import {MemoryRouter} from 'react-router-dom';
import BlogSideBar from './BlogSideBar';

describe("BlogSideBar", () => {
	let featuredPosts;
	let pathParams;
	let setRedirect;

	beforeEach(() => {
		featuredPosts = [];
		pathParams = [
			{name: "Strona główna", path: "qwe"},
			{name: "Strona 1", path: "asd"},
			{name: "Strona 2", path: "zxc"}
		];
		setRedirect = jest.fn();
		jest.clearAllMocks();
	});

	it("should match snapshot", () => {
		const {container} = render(
			<MemoryRouter>
				<BlogSideBar pathParams={pathParams} setRedirect={setRedirect} featuredPosts={featuredPosts} />
			</MemoryRouter>
		);

		expect(container).toMatchSnapshot();
	});

	it("should call setRedirect with correct params", () => {
		const {getByText} = render(
			<MemoryRouter>
				<BlogSideBar pathParams={pathParams} setRedirect={setRedirect} featuredPosts={featuredPosts} />
			</MemoryRouter>
		);

		fireEvent.click(getByText(pathParams[1].name));

		expect(setRedirect).toHaveBeenCalledWith({
			action: true,
			destination: pathParams[1].path
		})
	});
});