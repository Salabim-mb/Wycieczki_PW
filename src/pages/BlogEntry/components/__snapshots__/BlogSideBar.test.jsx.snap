// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogSideBar should match snapshot 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4"
  >
    <div
      class="MuiPaper-root makeStyles-sidebarBreadcrumbs-1 MuiPaper-elevation0 MuiPaper-rounded"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6"
      >
        Nawigacja
      </h6>
      <nav
        aria-label="breadcrumb"
        class="MuiTypography-root MuiBreadcrumbs-root MuiTypography-body1 MuiTypography-colorTextSecondary"
      >
        <ol
          class="MuiBreadcrumbs-ol"
        >
          <li
            class="MuiBreadcrumbs-li"
          >
            <div
              class="MuiButtonBase-root MuiChip-root MuiChip-outlined MuiChip-clickable"
              role="button"
              tabindex="0"
            >
              <span
                class="MuiChip-label"
              >
                Strona główna
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </div>
          </li>
          <li
            aria-hidden="true"
            class="MuiBreadcrumbs-separator"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
              />
            </svg>
          </li>
          <li
            class="MuiBreadcrumbs-li"
          >
            <div
              class="MuiButtonBase-root MuiChip-root MuiChip-outlined MuiChip-clickable"
              role="button"
              tabindex="0"
            >
              <span
                class="MuiChip-label"
              >
                Strona 1
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </div>
          </li>
          <li
            aria-hidden="true"
            class="MuiBreadcrumbs-separator"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
              />
            </svg>
          </li>
          <li
            class="MuiBreadcrumbs-li"
          >
            <div
              class="MuiButtonBase-root MuiChip-root MuiChip-outlined MuiChip-clickable"
              role="button"
              tabindex="0"
            >
              <span
                class="MuiChip-label"
              >
                Strona 2
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </div>
          </li>
        </ol>
      </nav>
    </div>
    <h6
      class="MuiTypography-root MuiTypography-h5"
    >
      Proponowane wpisy
    </h6>
    Tu będą ze 3 tytuły proponowanych / losowych wpisów
  </div>
</div>
`;
